src/main/java/uk/gov/laa/ccms/caab/client/AbstractApiClientErrorHandler.java

```diff
@@ -0,0 +1,145 @@
+package uk.gov.laa.ccms.caab.client;
+
+
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.util.MultiValueMap;
+import reactor.core.publisher.Mono;
+
+/**
+ * Provides error-handling capabilities for the API client interactions.
+ */
+@Slf4j
+public abstract class AbstractApiClientErrorHandler {
+
+  /**
+   * Creates an exception with a specified message and cause.
+   *
+   * @param message the detail message for the exception.
+   * @param cause the cause of the exception.
+   * @return a new instance of RuntimeException.
+   */
+  protected RuntimeException createException(final String message, final Throwable cause) {
```

Comment:
Do you not want to make this method abstract, so it has to be defined in the specific error handlers?

Reply:
makes sense, will amend